### PR TITLE
Document Access Model

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #:* /bcgov/entity#12414

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: search-api/src/search_api/models/document.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+"""Table for storing document details."""
+from enum import Enum
+
+from .db import db
+
+
+class Document(db.Model):
+    """Used to hold the documents requested in an access request."""
+
+    class DocumentTypes(str, Enum):
```
Comment (thorwolpert) [inline search-api/src/search_api/models/document.py:23]: I think this should use our new BaseEnum

File: search-api/src/search_api/models/document.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        CERTIFICATE_OF_GOOD_STANDING = 'cogs'
+        CERTIFICATE_OF_STATUS = 'cert_of_status'
+        LETTER_UNDER_SEAL = 'letter_under_seal'
+
+    __versioned__ = {}
+    __tablename__ = 'document'
+
+    id = db.Column(db.Integer, primary_key=True)
+    document_type = db.Column(db.String(30), index=True)
+    document_key = db.Column('document_key', db.String(100), nullable=False)
```
Comment (thorwolpert) [inline search-api/src/search_api/models/document.py:36]: Should this use the Enum subclassed from BaseEnum?

File: search-api/src/search_api/models/document.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    access_request_id = db.Column('access_request_id', db.Integer, db.ForeignKey('document_access_request.id'))
+
+    @property
+    def json(self):
+        """Return a dict of this object, with keys in JSON format."""
+        document = {
+            'id': self.id,
+            'documentType': self.document_type,
+            'documentKey': self.document_key,
+            'fileName': self.file_name or '',
```
Comment (thorwolpert) [inline search-api/src/search_api/models/document.py:48]: Is this better than setting it to _None_?

File: search-api/src/search_api/models/document_access_request.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    class Status(str, Enum):
+        """Enum of the request statuses."""
+
+        CREATED = 'CREATED'
+        PAID = 'PAID'
+        ERROR = 'ERROR'
+        COMPLETED = 'COMPLETED'  # After all docs are generated and stored.
+
+    __versioned__ = {}
+    __tablename__ = 'document_access_request'
```
Comment (thorwolpert) [inline search-api/src/search_api/models/document_access_request.py:40]: We aren't using versioned table here, as we're not including continuum.

**Review by thorwolpert — COMMENTED** [review]
couple of discussion points.

**Review by kialj876 — APPROVED** [review]
Looks good to me 👍

File: search-api/src/search_api/models/document.py (hunk 2 above)
Comment (lekshmimallika-aot) [reply search-api/src/search_api/models/document.py:37]: Changed to subclass from BaseEnum

File: search-api/src/search_api/models/document.py (hunk 1 above)
Comment (lekshmimallika-aot) [reply search-api/src/search_api/models/document.py]: Done

File: search-api/src/search_api/models/document.py (hunk 3 above)
Comment (lekshmimallika-aot) [reply search-api/src/search_api/models/document.py]: No :)

File: search-api/src/search_api/models/document_access_request.py (hunk 4 above)
Comment (lekshmimallika-aot) [reply search-api/src/search_api/models/document_access_request.py:40]: Removed.

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**